core/domain/user_jobs_one_off.py

```diff
@@ -324,3 +326,54 @@ def map(user_model):
     @staticmethod
     def reduce(key, stringified_values):
         pass
+
+
+class UserRolesMigrationOneOffJob(jobs.BaseMapReduceJobManager):
+    """A one time job to attach a new field (representing role of user) to the
+    UserSettingsModel. The job will load all existing users from the datastore,
+    look for the role to assign them and then store them back thus updating the
+    table schema and adding appropiate roles to users.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_model):
+        admin_usernames = config_domain.Registry.get_config_property(
+            'admin_usernames')
+        moderator_usernames = config_domain.Registry.get_config_property(
+            'moderator_usernames')
+        banned_usernames = config_domain.Registry.get_config_property(
+            'banned_usernames')
+        collection_editors = config_domain.Registry.get_config_property(
+            'collection_editor_whitelist')
+        email_senders = config_domain.Registry.get_config_property(
+            'whitelisted_email_senders')
+
+        try:
+            if user_model.username in admin_usernames.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_ADMIN)
+            elif user_model.username in moderator_usernames.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_MODERATOR)
+            elif user_model.username in banned_usernames.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_BANNED_USER)
+            elif user_model.username in collection_editors.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_COLLECTION_EDITOR)
+            elif user_model.username in email_senders.value:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_SUPER_ADMIN)
+            else:
+                user_services.update_user_role(
+                    user_model.id, feconf.ROLE_EXPLORATION_EDITOR)
+            yield(user_model.username, 'Role successfully attached.')
```

Comment:
I guess it's fine to yield something like ('Success', 1), and reduce on their sum. That way, you don't get a whole lot of distinct keys; currently, you get one key per username which seems unnecessary.

So one could do that, and sum the values in the success case, and concat the usernames in the error case. But that seems more complicated than just doing the latter and I don't really see the point of yielding in the success case anyway (I'm not sure what we'd use the data for).